packages/client/src/Ethereum.ts

```diff
@@ -31,24 +31,22 @@ const getRpcProviders = (connectionInfo: ChainConnectionInfo, pollInterval?: num
     })
 }
 
-export const getStreamRegistryOverrides = (config: Pick<StrictStreamrClientConfig, 'contracts'>): Overrides => {
-    const primaryProvider = getStreamRegistryChainProviders(config)[0]
-    return getOverrides(config.contracts.streamRegistryChainRPCs.name ?? 'polygon', primaryProvider, config)
-}
-
 /**
  * Apply the gasPriceStrategy to the estimated gas price, if given
  * Ethers.js will resolve the gas price promise before sending the tx
  */
-const getOverrides = (chainName: string, provider: Provider, config: Pick<StrictStreamrClientConfig, 'contracts'>): Overrides => {
-    const chainConfig = config.contracts.ethereumNetworks[chainName]
-    if (chainConfig === undefined) { return {} }
+export const getEthersOverrides = (config: Pick<StrictStreamrClientConfig, 'contracts'>): Overrides => {
+    const chainConfig = config.contracts.ethereumNetwork
     const overrides = chainConfig.overrides ?? {}
-    if (chainConfig.highGasPriceStrategy) {
-        const gasPriceStrategy = (estimatedGasPrice: BigNumber) => estimatedGasPrice.add('10000000000') 
+    if ((chainConfig.highGasPriceStrategy) && (chainConfig.overrides?.gasPrice === undefined)) {
+        const primaryProvider = getStreamRegistryChainProviders(config)[0]
```

Comment:
this feels a bit indirect. If we use highGasPriceStrategy, then we presumably use it in a network, and we should probably use the provider of that network, not necessarily "stream registry chain provider". I'd pass the provider as argument to this function the use it directly.

Typical use would look like so:
```
        const ethersOverrides = this.getEthersOverrides(this.operatorContract.provider)
        ethersOverrides.gasLimit = '200000'
        await (await this.operatorContract.voteOnFlag(
            sponsorship,
            targetOperator,
            voteData,
            ethersOverrides
        )).wait()
```
or if we have just one provider in use in that instance, use that.

Reply:
of course if we know we only work in one network/chain, then sure, let's keep it that way.